services/src/main/scala/cromwell/services/womtool/models/WorkflowDescription.scala

```diff
@@ -0,0 +1,113 @@
+package cromwell.services.womtool.models
+
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.deriveEncoder
+import wom.callable.Callable.{FixedInputDefinition, InputDefinition, InputDefinitionWithDefault}
+import wom.executable.WomBundle
+
+// Very provisional types for some of these, and perhaps the defaults will go away later in development
+case class WorkflowDescription(
+                                valid: Boolean,
+                                errors: List[String],
+                                name: String = "",
+                                inputs: List[InputDescription] = List.empty,
+                                outputs: List[OutputDescription] = List.empty,
+                                images: List[String] = List.empty,
+                                submittedDescriptorType: Map[String, String] = Map.empty,
+                                importedDescriptorTypes: List[Map[String, String]] = List.empty,
+                                meta: Map[String, String] = Map.empty
+                              )
+
+case object WorkflowDescription {
+
+  def withErrors(errors: List[String]): WorkflowDescription = {
+    WorkflowDescription(
+      valid = false,
+      errors = errors
+    )
+  }
+
+  def fromBundle(bundle: WomBundle, languageName: String, languageVersionName: String): WorkflowDescription = {
+
+    val sdt = Map(
+      "descriptorType" -> languageName,
+      "descriptorTypeVersion" -> languageVersionName
+    )
+
+    bundle.primaryCallable match {
+      case Some(callable) =>
+        val inputs = callable.inputs.sortBy(_.name) map { input: InputDefinition =>
+
+          input match {
+            // TODO: is there a cool kids way to do this?
+            // I tried `thing @ (_: FixedInputDefinition | _: InputDefinitionWithDefault)` but Scala does not know `thing` has a `default`
```

Comment:
missing common trait?

Reply:
This is a good thought, I am looking at whether this makes sense semantically in the various places we use these types

Reply:
Implemented.